manim/utils/units.py

```diff
@@ -0,0 +1,41 @@
+from functools import wraps
+
+from ..config import config
+
+PERCENT = "%"
+PIXELS = "px"
+MUNIT = "m"
+
+
+def _size_from_dimension(dim=0, pixels=False):
+    if dim == 0:
+        return config["frame_width"] if not pixels else config["pixel_width"]
+    elif dim == 1:
+        return config["frame_height"] if not pixels else config["pixel_height"]
+    else:
+        # TODO: determine size in z-direction
+        raise ValueError("Dimension not supported.")
+
+
+class Unit():
+    def __init__(self, value, unit=None):
+        self._value = value
+        self._unit = unit
+
+    def value(self, dim=0):
```

Comment:
Yes and no, yes because it would be cleaner, no because you can't pass a dimension then. And the value depends on the dimension. Thus it has to be a function all as far as I'm aware.

Reply:
Oh. Good point. Now I'm not sure we should support different dimensions in the same class. How about this (this might be far-fetched..)

```python
def unit(self, value, unit, dim=0):
        if unit == MUnit:
            return Unit(value)
        elif unit == PIXELS:
            return PixelUnit(value)
        elif unit == PERCENT:
            return PercentUnit(value, dim)

class PercentUnit(Unit):
    # specific code for handling percent/relative units here

class PixelUnit(Unit):
    # specific code for handling pixels here
```